pkg/streamer/reader.go

```diff
@@ -562,6 +501,20 @@ func (r *BinlogReader) parseFile(
 			latestPos = int64(e.Header.LogPos)
 		}
 
+		// align with MySQL
+		// if an event's gtid has been contained by given gset
+		// replace it with HEARTBEAT event
+		// for Mariadb, it will bee replaced with MARIADB_GTID_LIST_EVENT
+		// In DM, we replace both of them with HEARTBEAT event
+		if r.replaceWithHeartbeat {
+			switch e.Event.(type) {
+			case *replication.RowsEvent, *replication.QueryEvent, *replication.GTIDEvent, *replication.XIDEvent, *replication.TableMapEvent:
+				// replace with heartbeat event
+				e = event.GenHeartbeatEvent(e.Header)
+			default:
```

Comment:
what's the skipped event type and why 🤔

Reply:
Other events like format_description_event,rotate_event should not be replaced.

Reply:
👍 please add a comment for it